packages/toast/src/ToastContext.story.tsx

```diff
@@ -0,0 +1,192 @@
+  },
+  args: {
+    /// @ts-expect-error
+    darkMode: false,
+  },
+});
+
+const BasicChildren = (props: Partial<InternalToastProps>) => {
+  const { pushToast, clearStack } = useToast();
+
+  return (
+    <div>
+      <div
+        className={css`
+          display: flex;
+          gap: 8px;
+        `}
+      >
+        <Button
+          data-testid="toast-trigger"
+          onClick={() => {
+            const variant = props.variant || sample(Variant);
+            pushToast({
+              title: `I'm a ${variant} toast`,
+              description: faker.lorem.lines(random(1, 2)),
+              variant,
+              ...props,
+            });
+          }}
+        >
+          Push toast
+        </Button>
+        <Button onClick={() => clearStack()}>Clear all</Button>
+      </div>
+      {/* TODO: Remove this */}
+      <Body>
+        Please provide any feedback in{' '}
+        <Link href="https://docs.google.com/document/d/1xgzfJhkWugpEeBlbomTpkrlbjedNxdwn7A0vg3jvDXo/edit#">
+          this doc
+        </Link>
+      </Body>
+    </div>
+  );
+};
+
+export const Basic = ({
+  darkMode,
+  ...props
+}: Partial<InternalToastProps> & DarkModeProps) => {
+  return (
+    <LeafyGreenProvider darkMode={darkMode}>
+      <ToastProvider>
+        <BasicChildren {...props} />
+      </ToastProvider>
+    </LeafyGreenProvider>
+  );
+};
+
+const VariantsChildren = (props: Partial<InternalToastProps>) => {
+  const { pushToast, clearStack, getStack, updateToast } = useToast();
+
+  const stack = getStack();
+  const progressToasts =
+    stack && Array.from(stack).filter(([_, t]) => t.variant === 'progress');
+
+  const [progress, setProgress] = useState(0);
+
+  useEffect(() => {
+    if (!progressToasts || progressToasts.length === 0) {
+      setProgress(0);
+    }
+  }, [progressToasts]);
+
+  return (
+    <div>
+      <div
+        className={css`
+          display: grid;
+          grid-template-columns: repeat(3, 1fr);
+          gap: 8px;
+          margin: 1em 0;
+        `}
+      >
+        {Object.values(Variant).map(variant => {
+          const VariantIcon = variantIcons[variant];
+
+          return (
+            <Button
+              key={variant}
+              onClick={() => {
+                pushToast({
+                  title: `I'm a ${variant} toast`,
+                  description: faker.lorem.lines(random(1, 2)),
+                  variant,
+                  progress: 0,
+                  ...props,
+                });
+              }}
+            >
+              <VariantIcon /> {startCase(variant)} toast
+            </Button>
+          );
+        })}
+        <Button onClick={() => clearStack()}>Clear all</Button>
+      </div>
+      {progressToasts && progressToasts.length > 0 && (
+        <>
+          <Label htmlFor="progress">Progress: </Label>
+          <InlineCode>{progress}</InlineCode>
+          <br />
+          <input
+            value={progress}
+            onChange={e => {
```

Comment:
should we be defining this not inline

Reply:
Technically yes, but it's in the story, so I'm less worried about this one